sdk/core/azure-core/inc/azure/core/datetime.hpp

```diff
@@ -117,14 +117,14 @@ class DateTime final : public _detail::Clock::time_point {
    * @brief Constructs an instance of `%DateTime` from a `time_point`.
    *
    */
-  constexpr DateTime(time_point const& timePoint) : time_point(timePoint) {}
+  constexpr DateTime(time_point const& timePoint) noexcept : time_point(timePoint) {}
 
   /**
    * @brief Construct an instance of `%DateTime` from `std::chrono::system_clock::time_point`.
    * @param systemTime A value of `std::chrono::system_clock::time_point`.
    *
    */
-  DateTime(std::chrono::system_clock::time_point const& systemTime)
+  DateTime(std::chrono::system_clock::time_point const& systemTime) noexcept
```

Comment:
This is a risky path where we might never be able reach an ending.  We are calling `time_since_epoch()` inside this function, which is not `noexcept` and in theory, might throw right?

Should we explore removing `noexcept` instead?  As it is not really compile-time enforced, so it becomes a fragile structure built only by developers looking at the code or looking at weird errors where developers can't catch an exception.

We need to define better guidelines for using `noexcept` only for those methods which might be benefit for the performance optimization and which don't depend on other method calls....

@LarryOsterman @CaseyCarter  @ahsonkhan   thoughts ?

Reply:
I know, but I checked with MSVC's headers, everything we call is `noexcept`. So I made this a draft PR so far to run this through CI to see if it would trigger any errors.
Unfortunately, I can't find mentions that the C++ standard require the chrono functions that we call to be `noexcept` (compilers just go beyond).
So, I'll make a catch all instead.